### PR TITLE
feat(PositionManager): Add totalStakedTokenAmount

### PR DESCRIPTION
**Pull request by Woody810**
### Summary
📤📁💰

The pull request adds a feature to display the total value of the staked assets in the second token of the pair in the `PositionManagers` view. It creates and exports a custom hook `useTotalAssetInStaked1Token` and uses it in the `StakedAssets` component.

> _`StakedAssets` shows_
> _total value in second token_
> _autumn harvest time_

### Walkthrough
*  Add a hook to calculate the total value of the staked assets in terms of the second token in the pair ([link](https://github.com/pancakeswap/pancake-frontend/pull/8298/files?diff=unified&w=0#diff-0f9f1358ea4639321919c0fc4d4a63eea2814d63927fecb2b704fe1e3ac58530R1-R18))
*  Export the hook from the `hooks/index.ts` file ([link](https://github.com/pancakeswap/pancake-frontend/pull/8298/files?diff=unified&w=0#diff-7f97336d46156ac9f46cd2613ef4448613b82774ff0f1569c5497b020aa23d37R11))
*  Import the hook and display the result in the `StakedAssets` component ([link](https://github.com/pancakeswap/pancake-frontend/pull/8298/files?diff=unified&w=0#diff-a50beaeb766e2af9a03a6593bb5b1490b26b3299a89e659465bb8d850df32654L9-R9), [link](https://github.com/pancakeswap/pancake-frontend/pull/8298/files?diff=unified&w=0#diff-a50beaeb766e2af9a03a6593bb5b1490b26b3299a89e659465bb8d850df32654R46-R51), [link](https://github.com/pancakeswap/pancake-frontend/pull/8298/files?diff=unified&w=0#diff-a50beaeb766e2af9a03a6593bb5b1490b26b3299a89e659465bb8d850df32654R66-R68))

telegram-cloud-photo-size-5-6215065578985600594-x

### CODE REVIEW

File: apps/web/src/views/PositionManagers/hooks/useTotalAssetInSingleDepositTokenAmount.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  singleDepositTokenAmount?: CurrencyAmount<Currency>,
+  ohterTokenAmount?: CurrencyAmount<Currency>,
+  singleDepositTokenPriceUSD?: number,
+  ohterTokenPriceUSD?: number,
+) => {
+  const totalAssetInSingleDepositTokenAmount = useMemo(() => {
+    return (
+      Number(formatAmount(singleDepositTokenAmount) ?? 0) +
+      (Number(formatAmount(ohterTokenAmount) ?? 0) * (ohterTokenPriceUSD ?? 0)) / (singleDepositTokenPriceUSD ?? 0)
+    )
```
Comment (ChefMomota) [inline apps/web/src/views/PositionManagers/hooks/useTotalAssetInSingleDepositTokenAmount.ts:15]: For the number calculation can u use `BigNumber` to do it?

Comment (Woody810) [reply apps/web/src/views/PositionManagers/hooks/useTotalAssetInSingleDepositTokenAmount.ts:15]: got it, will fix it later

File: apps/web/src/views/PositionManagers/components/StakedAssets.tsx [hunk 2]
```diff
@@ -57,6 +71,10 @@ export const StakedAssets = memo(function StakedAssets({
           <Text color="text" fontSize="1.5em" bold>
             ~${totalAssetsInUsd.toFixed(2)}
           </Text>
+          <Text color="textSubtle" fontSize="0.75em">
+            (~{totalAssetInSingleDepositTokenAmount.toFixed(6)}{' '}
+            {isSingleDepositToken0 ? currencyA.symbol : currencyB.symbol})
```
Comment (ChefMomota) [inline apps/web/src/views/PositionManagers/components/StakedAssets.tsx:76]: Can we make it like this, can use useMemo show the final text.

const symbol = isSingleDepositToken0 ? currencyA.symbol : currencyB.symbol}
{~`${totalAssetInSingleDepositTokenAmount.toFixed(6)}} ${symbol}`}

Comment (Woody810) [reply apps/web/src/views/PositionManagers/components/StakedAssets.tsx]: Fixed.

File: apps/web/src/views/PositionManagers/hooks/useTotalAssetInSingleDepositTokenAmount.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  singleDepositTokenAmount?: CurrencyAmount<Currency>,
+  ohterTokenAmount?: CurrencyAmount<Currency>,
+  singleDepositTokenPriceUSD?: number,
+  ohterTokenPriceUSD?: number,
+) => {
+  const totalAssetInSingleDepositTokenAmount = useMemo(() => {
+    const singleDepositTokenAmountBigNum = new BigNumber(formatAmount(singleDepositTokenAmount) ?? BIG_ZERO)
+    const singleDepositTokenPriceUSDBigNum = new BigNumber(singleDepositTokenPriceUSD ?? BIG_ZERO)
+    const otherTokenAmountBigNum = new BigNumber(formatAmount(ohterTokenAmount) ?? BIG_ZERO)
+    const ohterTokenPriceUSDBigNum = new BigNumber(ohterTokenPriceUSD ?? BIG_ZERO)
```
Comment (ChefMomota) [inline apps/web/src/views/PositionManagers/hooks/useTotalAssetInSingleDepositTokenAmount.ts:17]: Remove `BIG_ZERO`, replace to `0`

File: apps/web/src/views/PositionManagers/components/StakedAssets.tsx [hunk 4]
```diff
@@ -39,11 +40,25 @@ export const StakedAssets = memo(function StakedAssets({
...
 }: StakedAssetsProps) {
   const { t } = useTranslation()
 
   const totalAssetsInUsd = useTotalAssetInUsd(staked0Amount, staked1Amount, token0PriceUSD, token1PriceUSD)
 
+  const singleDepositSymbol = isSingleDepositToken0 ? currencyA.symbol : currencyB.symbol
+  const singleDepositTokenAmount = isSingleDepositToken0 ? staked0Amount : staked1Amount
+  const singleDepositTokenPriceUSD = isSingleDepositToken0 ? token0PriceUSD : token1PriceUSD
+  const otherTokenAmount = isSingleDepositToken0 ? staked1Amount : staked0Amount
+  const otherTokenPriceUSD = isSingleDepositToken0 ? token1PriceUSD : token0PriceUSD
```
Comment (ChefMomota) [inline apps/web/src/views/PositionManagers/components/StakedAssets.tsx:53]: Use useMemo

**Review by ChefMomota — APPROVED** [review]

**Pull request merged**